src/main/app/src/utils/toteutus/toteutusRemoteErrorsToFormErrors.ts

```diff
@@ -106,20 +107,48 @@ export const toteutusRemoteErrorsToFormErrors: RemoteErrorsToFormErrors = ({
     };
   }
 
-  const opintojaksoIndex = path.match(/liitetytOpintojaksot\[(\d+)\]/)?.[1];
   if (
-    /metadata.liitetytOpintojaksot\[\d+]\.julkaisutila/.test(path) &&
+    /metadata.liitetytOpintojaksot.julkaisutila/.test(path) &&
     errorType === 'invalidTilaForLiitettyOpintojaksoOnJulkaisu'
   ) {
+    const liitetytOpintojaksot =
+      formValues?.opintojaksojenLiittaminen?.opintojaksot || [];
+    const indicesForOpintojaksotWithInvalidTila = liitetytOpintojaksot.reduce(
+      (invalidOpintojaksoIndices, { opintojakso }, index) => {
+        const indexOfInvalidOpintojakso = _.includes(
+          meta.toteutukset,
+          opintojakso.value
+        )
+          ? index
+          : null;
+
+        if (_.isNull(indexOfInvalidOpintojakso)) {
+          return invalidOpintojaksoIndices;
+        }
+
+        return [...invalidOpintojaksoIndices, indexOfInvalidOpintojakso];
+      },
+      []
+    );
```

Comment:
Tän vois ehkä tehdä näin vähän lyhyemmin?
```suggestion
    const indicesForOpintojaksotWithInvalidTila = _.map(meta.toteutukset, oid =>
      _.findIndex(liitetytOpintojaksot, ['opintojakso.value', oid])
    ).filter(i => i >= 0);
```